### PR TITLE
TreasureChest item retrieval

### PR DESCRIPTION
**Pull request by sbrad044**
Added a bunch of classes relating to interacting with chests. Also made a basic version of WalkingCharacter, InventoryInteractor, and TextGenerator. Deleted the old redundant HealthPotion.

### CODE REVIEW

**Comment by sbrad044** [comment]
this is missing InventoryInterator, which is in Siddarth's pull request, so it won't make sense by iteself

File: src/test/java/ChestTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @Test
+    void alreadyOpenChestTest() {
+        TreasureChest chest = new TreasureChest(new HealthPotion(), coordinates1);
+        chest.open();
+        assertEquals(true, chest.isOpened());
+        assertEquals(false, chest.open());
+    }
+}
```
Comment (hoodiexxx) [inline src/test/java/ChestTest.java:33]: Good test and I believe it cover all the corner cases.

**Review by hoodiexxx — APPROVED** [review]

**Review by siddharthgowda — CHANGES_REQUESTED** [review]
Most of these classes are not entities even though they are part of the enitity package (i.e. chestItem getter and chestlooper but there are some more). Entities are really the model of our application. If something does not really store data it is not really an entitiy. Many of those classes are single function class (which are apprioriate in many casses) but here it might be better to but them all together in some sort of interactor.

Overall, there is quite a bit of refactoring to do. The idea behind the logic seems right although there are some syntactical things to fix like referencing GameState without actually having an instance of it in the class. (this might not matter if we end up making GameState more like a static class)

**Review by m-grigoriu — COMMENTED** [review]
Everything looks good.
I think WalkingCharacter should eventually have a 'sprite' attribute, but I am not sure if that is how we want to implement it.

**Comment by sbrad044** [comment]
> Many of those classes are single function class (which are apprioriate in many casses) but here it might be better to but them all together in some sort of interactor.

So wait are you saying I should make an interactor to oversee all these classes? Or combine them all into one interactor?

**Comment by sbrad044** [comment]
> Many of those classes are single function class (which are apprioriate in many casses) but here it might be better to but them all together in some sort of interactor.

So wait are you saying I should make an interactor to oversee all these classes? Or combine them all into one interactor?

**Comment by siddharthgowda** [comment]
> > Many of those classes are single function class (which are apprioriate in many casses) but here it might be better to but them all together in some sort of interactor.
> 
> So wait are you saying I should make an interactor to oversee all these classes? Or combine them all into one interactor?

This is honestly a hard question to answer because it depends. If you have a central theme around a group of methods and those methods are dependent on each other (not in terms of implementation) to get and updating the information in an entity then put them all in one class. If you think one of the methods is really just an implementation details and it's better to let the core "interactor" delegate work to that method, make that method its own class and the interactor class has that delegate class.
